pysindy/optimizers/frols.py

```diff
@@ -173,7 +173,7 @@ def _reduce(self, x, y):
             )
 
         # History of selected functions: [iteration x output x coefficients]
-        self.history_ = np.zeros((n_features, n_targets, n_features), dtype=x.dtype)
+        self.history_ = [np.zeros((n_features, n_targets, n_features), dtype=x.dtype)]
```

Comment:
The type of `history_` should be:

```python
self.history_: list[np.ndarray[tuple[N_Targets, N_Features], np.dtype[np.floating[np.typing.NBitBase]]]]
```

The point is that each entry in the list is the value of the coefficients at a particular iteration